src/rapids_metadata/__init__.py

```diff
@@ -12,55 +12,88 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-from os import PathLike
+from . import metadata as md
```

Comment:
Personal preference, do what you want with it, but in my opinion it'd be better to instead do this:

```python
from .metadata import (
    RAPIDSMetadata,
    RAPIDSRepo,
    RAPIDSVersion,
)
```

Those names are already sufficiently RAPIDS-y to avoid conflicts with other things, and that'd remove the need to repeat `md.` everywhere those classes are used.

This one is a minor stylistic preference though, feel free to ignore it.